src/go/rpk/pkg/cli/cmd/debug/bundle.go

```diff
@@ -0,0 +1,842 @@
+	ReplFactor    int16              `json:"replFactor"`
+}
+
+type consumerGroup struct {
+	Id      string                 `json:"id"`
+	Members []*consumerGroupMember `json:"members"`
+}
+
+type consumerGroupMember struct {
+	Host string `json:"host"`
+	Id   string `json:"id"`
+}
+
+func saveKafkaMetadata(ps *stepParams, admin sarama.ClusterAdmin) step {
+	return func() error {
+		log.Debug("Reading Kafka metadata")
+		bs, _, err := admin.DescribeCluster()
+		if err != nil {
+			return fmt.Errorf("couldn't list topics: %w", err)
+		}
+
+		brokers := make([]*broker, 0, len(bs))
+		for _, b := range bs {
+			brokers = append(brokers, &broker{
+				Id:   b.ID(),
+				Addr: b.Addr(),
+			})
+		}
+
+		topicsMap, err := admin.ListTopics()
+		if err != nil {
+			return fmt.Errorf("couldn't list topics: %w", err)
+		}
+		topics := make([]*topic, 0, len(topicsMap))
+		for topicName, details := range topicsMap {
+			n, ds := topicName, details
+			topics = append(topics, &topic{
+				Name:          n,
+				Config:        ds.ConfigEntries,
+				NumPartitions: ds.NumPartitions,
+				ReplFactor:    ds.ReplicationFactor,
+			})
+		}
+		cgs2proto, err := admin.ListConsumerGroups()
+		if err != nil {
+			return fmt.Errorf("couldn't list consumer groups: %w", err)
+		}
+		cgNames := make([]string, 0, len(cgs2proto))
+		for name := range cgs2proto {
+			cgNames = append(cgNames, name)
+		}
+		cgDescs, err := admin.DescribeConsumerGroups(cgNames)
+		if err != nil {
+			if len(cgDescs) == 0 {
+				return fmt.Errorf("couldn't list describe consumer groups: %w", err)
+			}
+		}
+		cgs := make([]*consumerGroup, 0, len(cgDescs))
+		for _, desc := range cgDescs {
+			d := desc
+			members := make([]*consumerGroupMember, 0, len(d.Members))
+			for _, m := range d.Members {
+				members = append(members, &consumerGroupMember{
+					Host: m.ClientHost,
+					Id:   m.ClientId,
+				})
+			}
+			cgs = append(cgs, &consumerGroup{
+				Id:      d.GroupId,
+				Members: members,
+			})
+		}
+		bytes, err := json.Marshal(&kafkaMetadata{
+			Brokers:        brokers,
+			Topics:         topics,
+			ConsumerGroups: cgs,
+		})
+		if err != nil {
+			return fmt.Errorf("couldn't marshal Kafka metadata: %w", err)
+		}
+		return writeFileToZip(ps, "kafka.json", bytes)
+	}
+}
+
+// Walks the redpanda data directory recursively, and saves to the bundle
+// a JSON map where the keys are the file/ dir paths, and the values are
+// objects containing their data: size, mode, the file or dir it points to
+// if the current file is a symlink, the time it was modified, its owner and
+// its group, as well as an error message if reading that specific file failed.
+func saveDataDirStructure(ps *stepParams, conf *config.Config) step {
+	type fileInfo struct {
+		Size     string    `json:"size"`
+		Mode     string    `json:"mode"`
+		Symlink  string    `json:"symlink,omitempty"`
+		Error    string    `json:"error,omitempty"`
+		Modified time.Time `json:"modified"`
```

Comment:
json isn't great at encoding a time.Time; we may want to `.Format` this.

Reply:
Done